src/Filter/StringFilter.php

```diff
@@ -29,13 +29,14 @@ class StringFilter extends Filter
 
     public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
     {
-        if (!$data || !\is_array($data) || !\array_key_exists('value', $data) || null === $data['value']) {
+        if (!$data || !\is_array($data) || !\array_key_exists('value', $data)) {
             return;
         }
 
-        $data['value'] = trim($data['value']);
```

Comment:
I'm not sure this can be done like this and is not considered as a BC break.
WDYT @sonata-project/contributors ?

What I meant was to create a boolean option `trim`, default to true.
And then we could do
```
$value = $data['value'] ?? '';
if ($this->getOption('trim')) {
    $value = trim($value);
}
```

But we definitely should do something about value with only spaces.

Before, if you were looking for `contains`: `space-space-space Vincent`, the extra-space were trimmed. I think it was useful, because we generally looking for a string, and extra space were not what we were searching. Plus it was an easy typing mistake.

Now, if you look for `contains`: ``, you looking for empty string.
But if you're looking for `contains`: `space-space-space`, I find weird that we're looking for empty string, and not especially three consecutive spaces.

Then, to me, the check
```
if ('' === $data['value'] && empty($data['type'])) {
```
is useless and can be removed, because it's already done by SonataAdmin.

Reply:
I added modes for trim.

Reply:
>  Then, to me, the check
> 
> ```
> if ('' === $data['value'] && empty($data['type'])) {
> ```
> is useless and can be removed, because it's already done by SonataAdmin.

Not certainly in that way. We only ignore cases when both the type and value of the filter are not specified. Now the SonataAdmin is really ignoring these cases, but PR https://github.com/sonata-project/SonataAdminBundle/pull/6402 will be change this behavior.

Reply:
The SonataAdmin Pr: https://github.com/sonata-project/SonataAdminBundle/pull/6402/files
is saying "If there is a value OR a type, use the filter".
So it means that if there is no value and no type, the filter is ignored.

The check 
```
if ('' === $data['value'] && empty($data['type'])) {
```
is saying "If there is no value and no type, ignore this".

To me, this is already excluded.

Reply:
IMO we should not trim string filter. What when someone want find double spaces in posts or something simmilar?

Reply:
I agree that the trim is questionable but it's the current behavior.

The only BC path I see would be to add the trim option in 3.x, and change the default value in 4.0 to TRIM_NONE, if we decide that's better to not trim.